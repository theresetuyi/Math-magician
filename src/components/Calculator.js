import React, { useState } from 'react';
import calculate from '../logic/calculate';

function Calculator() {
  const [state, setState] = useState({
    total: 0,
    next: null,
    operation: null,
  });

  const clickListener = (e) => {
    const result = calculate(state, e.target.value);
    setState(result);
  };

  const { total, next, operation } = state;
  const result = `${total || ''}${operation || ''}${next || ''}`;

  return (
    <div>
      <div className="header">Lets do some Math!</div>
      <form action="#">
        <div className="text">{result || '0'}</div>
        <div className="calc-btn">
          <input type="button" value="AC" onClick={clickListener} />
          <input type="button" value="+/-" onClick={clickListener} />
          <input type="button" value="%" onClick={clickListener} />
          <input type="button" value="รท" onClick={clickListener} className="sign" />
          <input type="button" value="7" onClick={clickListener} />
          <input type="button" value="8" onClick={clickListener} />
          <input type="button" value="9" onClick={clickListener} />
          <input type="button" value="x" onClick={clickListener} className="sign" />
          <input type="button" value="4" onClick={clickListener} />
          <input type="button" value="5" onClick={clickListener} />
          <input type="button" value="6" onClick={clickListener} />
          <input type="button" value="-" onClick={clickListener} className="sign" />
          <input type="button" value="1" onClick={clickListener} />
          <input type="button" value="2" onClick={clickListener} />
          <input type="button" value="3" onClick={clickListener} />
          <input type="button" value="+" onClick={clickListener} className="sign" />
          <input type="button" value="0" onClick={clickListener} className="zero" />
          <input type="button" value="." onClick={clickListener} />
          <input type="button" value="=" onClick={clickListener} className="sign" />
        </div>
      </form>
    </div>
  );
}

export default Calculator;
