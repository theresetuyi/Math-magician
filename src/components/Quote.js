import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

const Quote = ({ category }) => {
  const [quote, setQuote] = useState('');
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);

  useEffect(() => {
    const fetchQuote = async () => {
      try {
        setLoading(true);
        const url = 'https://api.api-ninjas.com/v1/quotes';
        const keys = 'b1PZcSwRaKAwryutCkkH2A==lVNi0KrhEV3io1wO';
        const response = await fetch(url, {
          headers: { 'X-Api-Key': keys },
          contentType: 'application/json',
        });
        const data = await response.json();
        if (data.length > 0) {
          setQuote(data[0].quote);
        }

        setLoading(false);
      } catch (error) {
        setError(false);
      }
    };
    fetchQuote();
  }, [category]);

  if (loading) {
    return <p>Loading...</p>;
  }

  if (error) {
    return <p>Error fetching quote</p>;
  }

  return (
    <div data-testId="quote-container">
      <p>{quote}</p>

    </div>
  );
};

Quote.propTypes = {
  category: PropTypes.string.isRequired,
};

export default Quote;
