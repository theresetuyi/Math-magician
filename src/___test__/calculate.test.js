import renderer from 'react-test-renderer';
import calculate from '../logic/calculate';
import Calculator from '../components/calculator';
import Home from '../components/Home';
import Quote from '../components/Quote';

describe('Test calculate', () => {
  it('6 + 4 = 10', () => {
    const obj = {
      total: '6',
      next: '4',
      operation: '+',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('10');
  });

  it('6 - 4 = 2', () => {
    const obj = {
      total: '6',
      next: '4',
      operation: '-',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('2');
  });

  it('6 x 4 = 24', () => {
    const obj = {
      total: '6',
      next: '4',
      operation: 'x',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('24');
  });

  it('6 รท 4 = 1.5', () => {
    const obj = {
      total: '6',
      next: '4',
      operation: 'รท',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('1.5');
  });

  it('6 % 4 = 2', () => {
    const obj = {
      total: '6',
      next: '4',
      operation: '%',
    };
    const res = calculate(obj, '=');
    expect(res.total).toBe('2');
  });
});
describe('Component UI render as expected', () => {
  it('Calculator renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Home renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Quote renders correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
