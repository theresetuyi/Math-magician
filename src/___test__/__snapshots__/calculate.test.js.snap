// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component UI render as expected Calculator renders correctly 1`] = `
<div>
  <div
    className="header"
  >
    Lets do some Math!
  </div>
  <form
    action="#"
  >
    <div
      className="text"
    >
      0
    </div>
    <div
      className="calc-btn"
    >
      <input
        onClick={[Function]}
        type="button"
        value="AC"
      />
      <input
        onClick={[Function]}
        type="button"
        value="+/-"
      />
      <input
        onClick={[Function]}
        type="button"
        value="%"
      />
      <input
        className="sign"
        onClick={[Function]}
        type="button"
        value="÷"
      />
      <input
        onClick={[Function]}
        type="button"
        value="7"
      />
      <input
        onClick={[Function]}
        type="button"
        value="8"
      />
      <input
        onClick={[Function]}
        type="button"
        value="9"
      />
      <input
        className="sign"
        onClick={[Function]}
        type="button"
        value="x"
      />
      <input
        onClick={[Function]}
        type="button"
        value="4"
      />
      <input
        onClick={[Function]}
        type="button"
        value="5"
      />
      <input
        onClick={[Function]}
        type="button"
        value="6"
      />
      <input
        className="sign"
        onClick={[Function]}
        type="button"
        value="-"
      />
      <input
        onClick={[Function]}
        type="button"
        value="1"
      />
      <input
        onClick={[Function]}
        type="button"
        value="2"
      />
      <input
        onClick={[Function]}
        type="button"
        value="3"
      />
      <input
        className="sign"
        onClick={[Function]}
        type="button"
        value="+"
      />
      <input
        className="zero"
        onClick={[Function]}
        type="button"
        value="0"
      />
      <input
        onClick={[Function]}
        type="button"
        value="."
      />
      <input
        className="sign"
        onClick={[Function]}
        type="button"
        value="="
      />
    </div>
  </form>
</div>
`;

exports[`Component UI render as expected Home renders correctly 1`] = `
<main>
  <h2>
    Welcome to the homepage!
  </h2>
  <p
    className="welcome"
  >
     
    lorem Udemy is a global marketplace for online courses of all shapes and sizes. From data science to graphic design and social media marketing, there are thousands of programs available for learning and teaching online. By enabling people to study from the comfort of their own homes, Udemy strives to help individuals enhance their skills and realize their visions. While users are able to individually sign up for courses, this particular landing page promotes Udemy’s business plan. It offers companies a platform which they can train and educate their employees, providing them with extra capabilities and on a range of topics. This can help businesses support their employees while working
  </p>
</main>
`;

exports[`Component UI render as expected Quote renders correctly 1`] = `
<div>
  <p />
</div>
`;
